src/test/java/uk/gov/hmcts/reform/sendletter/jupiter/GreenMailExtension.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.sendletter.jupiter;
+
+import com.icegreen.greenmail.configuration.GreenMailConfiguration;
+import com.icegreen.greenmail.util.GreenMail;
+import com.icegreen.greenmail.util.GreenMailProxy;
+import com.icegreen.greenmail.util.ServerSetup;
+import com.icegreen.greenmail.util.ServerSetupTest;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * This is a copy of not yet released code.
+ * {@see https://github.com/greenmail-mail-test/greenmail/tree/master/greenmail-junit5/src/main/java/com/icegreen/greenmail/junit5}
+ */
+public class GreenMailExtension extends GreenMailProxy implements BeforeEachCallback, AfterEachCallback {
```

Comment:
Do we really need this ? can we not use `GreenMailRule` instead which we used in bulkscan ?

Reply:
there's a docblock explaining this @nitinprabhu :D. right there one line above 😆

Reply:
What I meant was we can use it once it is released ?

Reply:
Yes, once it is released - definitely. Now we have to dump custom extension ourselves into codebase ;)